tests/functional/ext/dict_init_mutate.py

```diff
@@ -0,0 +1,33 @@
+"""Example cases for dict-init-mutate"""
+# pylint: disable=use-dict-literal, invalid-name
+
+base = {}
+
+fruits = {}
+for fruit in ["apple", "orange"]:
+    fruits[fruit] += 1
+
+count = 10
+fruits = {"apple": 1}
+fruits["apple"] += count
+
+config = {}  # [dict-init-mutate]
+config['dir'] = 'bin'
+config['user'] = 'me'
+config['workers'] = 5
+
+config = {}  # Not flagging calls to update for now
+config.update({"dir": "bin"})
+
+config = {}  # [dict-init-mutate]
+config['options'] = {}  # Identifying nested assignment not supporting this yet.
```

Comment:
I didn't give it much thought initially, but how difficult do you think it would be to catch this one?

Reply:
For another PR...